specification/cognitiveservices/OpenAI.Inference/models/completions/functions.tsp

```diff
@@ -69,3 +69,18 @@ model FunctionCall {
   """)
   arguments: string;
 }
+
+@doc("The name and arguments of a function that should be called, as generated by the model.")
+@added(ServiceApiVersions.v2023_07_01_Preview)
+model FunctionCallStreamDelta {
+  @doc("The name of the function to call.")
+  name?: string;
+
+  @doc("""
+  The arguments to call the function with, as generated by the model in JSON format.
+  Note that the model does not always generate valid JSON, and may hallucinate parameters
+  not defined by your function schema. Validate the arguments in your code before calling
+  your function.
+  """)
+  arguments?: string;
```

Comment:
should we make it a list of strings?

Reply:
Not in this case. This is the string is a essentially a JSON structure. It's the same as the one on line 70 if you look at the model `FunctionCall` the only difference that `FunctionCallStreamDelta` introduces is that since the fields could be segmented into different pieces by the SSE in the backend, we need to make everything optional.